### ISSUE TITLE
IrresponsibleModule triggers when a class defined in multiple files does not have descriptive comments for all files

### ISSUE THREAD

**Issue by beraboris** [troessner/reek#573]
When a class is defined in multiple files, IrresponsibleModule will trigger unless all the files have a descriptive comment for that class. This makes no sense. As long as one of the files has a descriptive comment, then it should be good enough.

Defining a class in multiple files is generally uncommon. There is one case where defining a class in multiple files is common.

Let's say you write a gem named `foobar`. In `lib/foobar.rb` you define the `Foobar` class. In `lib/foobar/stuff.rb` you define the `Foobar::Stuff` class. To define `Foobar::Stuff`, you need to reopen the `Foobar` class.

Here's an example:

```
$ ls
bar.rb  foo.rb
$ cat foo.rb 
# Some descritive comment
class Foobar
  # Foo has a comment
  class Foo; end
end
$ cat bar.rb 
class Foobar
  # bar has a comment
  class Bar; end
end
$ reek --version
reek 3.0.3
$ reek *
bar.rb -- 1 warning:
  [1]:Foobar has no descriptive comment (IrresponsibleModule)
1 total warning
```

**Comment by mvz** [comment]
Reek's architecture is not suitable for analysis across files. However, we could limit IrresponsibleModule to not trigger when a class or module is only there as a container for other classes or modules. I.e., no methods are defined in the class or module itself. So, the following would not trigger this smell anymore, regardless of `Foobar` being commented anywhere.

```ruby
class Foobar
  # Foo has a comment
  class Foo; end
end
```

**Comment by chastell** [comment]
> we could limit IrresponsibleModule to not trigger when a class or module is only there as a container for other classes or modules

:+1: – this is in line with ‘let’s not require comments for namespace-only modules’.

**Comment by mvz** [comment]
Fixed in #576.
